shared/menubar/files.desktop.js

```diff
@@ -46,14 +43,55 @@ const FileUpdate = (props: FileUpdateProps) => (
   </Kb.Box2>
 )
 
-const FileUpdates = ({updates, moreUpdateCount}: FileUpdatesProps) => (
+type FileUpdatesState = {
+  isShowingAll: boolean,
+}
+
+const FileUpdatesHoc = (ComposedComponent: React.ComponentType<any>) =>
+  class extends React.PureComponent<FileUpdatesProps, FileUpdatesState> {
+    state = {
+      isShowingAll: false,
+    }
+    render() {
+      return <ComposedComponent {...this.props} onShowAll={() => this.setState({isShowingAll: !this.state.isShowingAll})} isShowingAll={this.state.isShowingAll} />
+    }
+  }
```

Comment:
Hmm is there a reason you didn't just make `FileUpdatesShowAll` into a class and do these there? It seems clear that way.

Reply:
Yeah, could do that.

Reply:
I couldn't get it to work without a HOC.